builtin/game/item_entity.lua

```diff
@@ -200,28 +203,36 @@ core.register_entity(":__builtin:item", {
 						end
 					end
 				end
-				self.object:set_velocity({x = 0, y = 0, z = 0})
-				self.object:set_acceleration({x = 0, y = 0, z = 0})
-				self.physical_state = false
-				self.object:set_properties({physical = false})
+				-- Resting on a slippery node and possibly sliding
+				if slippery ~= 0 then
+					-- If horizontal velocity near zero, stop motion
+					if math.abs(v.x) < 0.1 and math.abs(v.z) < 0.1 then
+						self.object:set_velocity({x = 0, y = 0, z = 0})
+						self.object:set_acceleration({x = 0, y = 0, z = 0})
+						self.physical_state = false
+						self.object:set_properties({physical = false})
+					else -- Horizontal deceleration
+						local slip_factor = 4.0 / (slippery + 4)
+						self.object:set_acceleration({
+							x = -v.x * slip_factor,
+							y = 0,
+							z = -v.z * slip_factor
+						})
+					end
+				else -- Resting on a non-slippery node, stop motion
+					self.object:set_velocity({x = 0, y = 0, z = 0})
```

Comment:
Ok i'll leave this for SmallJoker to do since he seems happy to do it (?).
So can we merge this now?